src/main/java/org/apache/datasketches/quantilescommon/QuantilesGenericAPI.java

```diff
@@ -0,0 +1,349 @@
+   *
+   * <p>If the sketch is empty this returns null.</p>
+   *
+   * @param ranks the given array of normalized ranks, each of which must be
+   * in the interval [0.0,1.0].
+   * @param searchCrit if INCLUSIVE, the given ranks include all quantiles &le;
+   * the quantile directly corresponding to each rank.
+   * @return an array of quantiles corresponding to the given array of normalized ranks.
+   * @see org.apache.datasketches.quantilescommon.QuantileSearchCriteria
+   * @deprecated Use {@link #getQuantile(double, QuantileSearchCriteria)
+   * getQuantile(rank, searchCrit) in a loop.}
+   */
+  @Deprecated
+  T[] getQuantiles(double[] ranks, QuantileSearchCriteria searchCrit);
+
+  /**
+   * Same as {@link #getQuantiles(int, QuantileSearchCriteria) getQuantiles(numEvenlySpaced, INCLUSIVE)}
+   * @param numEvenlySpaced number of evenly spaced normalied ranks
+   * @return an array of quantiles that are evenly spaced by their ranks.
+   * @deprecated Use {@link #getQuantile(double, QuantileSearchCriteria)
+   * getQuantile(rank, searchCrit) in a loop.}
+   */
+  @Deprecated
+  default T[] getQuantiles(int numEvenlySpaced) {
+    return getQuantiles(numEvenlySpaced, INCLUSIVE);
+  }
+
+  /**
+   * This is a version of getQuantiles() where the caller only specifies the number of of desired evenly spaced,
+   * normalized ranks, and returns an array of the corresponding quantiles.
+   *
+   * <p>If the sketch is empty this returns null.
+   *
+   * @param numEvenlySpaced an integer that specifies the number of evenly spaced normalized ranks.
+   * This must be a positive integer greater than 0.
+   * <ul><li>Let <i>Smallest</i> and <i>Largest</i> be the smallest and largest quantiles
+   * retained by the sketch algorithm, respectively.
+   * (This should not to be confused with {@link #getMinItem} and {@link #getMaxItem},
+   * which are the smallest and largest quantiles of the stream.)</li>
+   * <li>A 1 will return the Smallest quantile.</li>
+   * <li>A 2 will return the Smallest and Largest quantiles.</li>
+   * <li>A 3 will return the Smallest, the Median, and the Largest quantiles.</li>
+   * <li>Etc.</li>
+   * </ul>
+   *
+   * @param searchCrit if INCLUSIVE, the given ranks include all quantiles &le; the quantile directly corresponding to
+   * each rank.
+   * @return an array of quantiles that are evenly spaced by their ranks.
+   * @see org.apache.datasketches.quantilescommon.QuantileSearchCriteria
+   */
+  T[] getQuantiles(int numEvenlySpaced, QuantileSearchCriteria searchCrit);
+
+  /**
+   * Same as {@link #getRank(Object, QuantileSearchCriteria) getRank(T quantile, INCLUSIVE)}
+   * @param quantile the given quantile
+   * @return the normalized rank corresponding to the given quantile
+   */
+  default double getRank(T quantile) {
+    return getRank(quantile, INCLUSIVE);
+  }
+
+  /**
+   * Gets the normalized rank corresponding to the given a quantile.
+   *
+   * <p>If the sketch is empty this returns null.</p>
```

Comment:
NaN?

Reply:
You are right null makes no sense.  But if the sketch is empty don't you think it should return zero ?  i.e., no mass?

Reply:
not sure about that. if the sketch is empty then nothing is known about the distribution. it is a matter of convention I suppose